app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt

```diff
@@ -74,50 +75,67 @@ class QuickViewModel(
         analyticsManager.trackScreen("QuickScreen")
 
         intent {
-            if (isRatesAvailable().not())
+            if (currencyRepo.isRatesAvailable().not()) {
+                reduce {
+                    state.copy(noInternet = true)
+                }
                 return@intent
+            }
 
-            AppSharedFlow.ShowAddedSnackbarQuick.flow.onEach { visuals ->
-                postSideEffect(QuickScreenEffect.ShowSnackbarAdded(visuals))
-            }.launchIn(viewModelScope)
-
-            quickRepo.allFlow().drop(1).onEach { quick ->
-                intent {
-                    val pages = mapPairsToPages(quick)
-                    reduce {
-                        state.copy(
-                            pages = pages
-                        )
-                    }
-                }
-            }.launchIn(viewModelScope)
+            init()
+        }
+    }
 
-            val allCurrencies = currencyRepo.getCurrencyNameUnsafe()
-            calcFrequentCurrUseCase.flow().drop(1).onEach {
-                val frequent = calcFrequentCurrUseCase.invoke()
-                    .map { currencyRepo.nameByCodeUnsafe(it) }
-                val topResults = getTopResultUseCase()
+    private fun init() = intent {
+        AppSharedFlow.ShowAddedSnackbarQuick.flow.onEach { visuals ->
+            postSideEffect(QuickScreenEffect.ShowSnackbarAdded(visuals))
+        }.launchIn(viewModelScope)
+
+        quickRepo.allFlow().drop(1).onEach { quick ->
```

Comment:
As these calling methods do not depend on each other, do you think we can make them run in parallel?
Also as we already have `intent { }` block outside, is it fine if we can remove one inside?
```kotlin
    private fun init() = intent {
        viewModelScope.launch {
            launch {
                AppSharedFlow.ShowAddedSnackbarQuick.flow.onEach { visuals ->
                    postSideEffect(QuickScreenEffect.ShowSnackbarAdded(visuals))
                }
            }

            launch {
                     intent {
                        val pages = mapPairsToPages(quick)
                        reduce {
                            state.copy(
                                pages = pages
                            )
                        }
                    }
            }

            launch {
                val allCurrencies = currencyRepo.getCurrencyNameUnsafe()
                calcFrequentCurrUseCase.flow().drop(1).onEach {
                    val frequent = calcFrequentCurrUseCase.invoke()
                        .map { currencyRepo.nameByCodeUnsafe(it) }
                    val topResults = getTopResultUseCase()
                    reduce {
                        state.copy(
                            frequent = frequent,
                            topResults = topResults
                        )
                    }
                }
                val frequent = calcFrequentCurrUseCase()
                    .map { currencyRepo.nameByCodeUnsafe(it) }
                val topResults = getTopResultUseCase()
                val pages = mapPairsToPages(quickRepo.getAll())
                reduce {
                    state.copy(
                        currencies = allCurrencies,
                        frequent = frequent,
                        topResults = topResults,
                        pages = pages,
                        initialized = true
                    )
                }
            }
        }
    }

```

Reply:
They are already launched in parallel due to `launchIn(viewModelScope)`
And these flows are not executed during initialization due to `drop(1)`

Reply:
Ohh thanks for explaination. Today I learn